aspnetcore/blazor/blazor-server-ef-core.md

```diff
@@ -0,0 +1,120 @@
+    Loading = false;
+    ```
+* For longer-lived operations that take advantage of EF Core's [change tracking](https://docs.microsoft.com/ef/core/querying/tracking) and/or [concurrency control](https://docs.microsoft.com/ef/core/saving/concurrency), scope the context to the lifetime of the component.
+
+### Create new DbContext instances
+
+The fastest way to create a new `DbContext` instance is by using `new` to create a new instance. There are several scenarios that may require additional dependencies to be resolved. For example, you may wish to use [DbContextOptions](https://docs.microsoft.com/ef/core/miscellaneous/configuring-dbcontext#configuring-dbcontextoptions) to configure the context. One approach is to register the options in DI and inject them into the component, like this:
+
+```razor
+@inject DbContextOptions<MyContext> Options
+
+@code {
+    public Task LoadAsync(int id)
+    {
+        using var context = new MyContext(Options);
+        return context.MyEntities.SingleOrDefaultAsync(e => e.Id == id);
+    }
+}
+```
+
+Another solution is to use a factory. EF Core 5.0 (preview) provides a built-in factory for creating new contexts. Because the sample app is based on the latest stable versions and not the preview, it implements its own factory in `Data/DbContextFactory.cs`. 
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs)]
+
+The factory is injected into components and used to create new instances. For example, in `Pages/Index.razor`:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/Index.razor?range=199-212)]
+
+### Scope to the component lifetime
+
+There are two approaches to scope a context to the component lifetime. If the context is registered with DI, you can use the special `OwningComponentBase` base class to create a control scope. Learn more about the class [here](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.components.owningcomponentbase-1).
+
+The second approach is to use the factory to create a context and track it for the lifetime of the component. First, implement `IDisposable` and inject the factory as shown in `Pages/EditContact.razor`.
+
+[!code-razor[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=5-7)]
+
+The sample app includes a helper method for disposal of the current context and ensures it is disposed when the component is:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=172-191)]
+
+Finally, `OnInitializedAsync` is overridden to create a new context. In the sample app, it loads the contact in the same method. The `DisposeContext` call ensures any previous context is disposed when the component is reused.
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=85-98)]
```

Comment:
This https://github.com/JeremyLikness/AspNetCore.Docs/blob/jrl-blazor-server-ef-core/aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor#L92 looks wrong, on initalizedasync only gets called once per component. Implementing dispose should be enough

Reply:
If you want to dispose the previous context when the component instance gets reused you are looking for OnParametersSetAsync

Reply:
Makes sense. I updated the changes and simplified the control.